pandas/core/indexes/api.py

```diff
@@ -89,13 +103,19 @@ def conv(i):
         index = indexes[0]
         for other in indexes[1:]:
             if not index.equals(other):
+
```

Comment:
isn't this already passed in as ``sort=True``?

Reply:
What do you mean?

Reply:
I mean this is deeply buried here, why is this showing a user facing warning? IOW what is the user call that triggers this?

Reply:
> IOW what is the user call that triggers this?

`pd.concat` or `DataFrame.append` with unaligned objects and an unspecified `sort`.

Where else would it go? Before this point we don't know what the indexes to align are, and whether or not they're aligned.